test/tandem/testTandemSimulator.js

```diff
@@ -387,81 +386,124 @@ describe('tandemSimulator.js', function() {
-      expectedSecondBasal.set('previous', expectedFirstBasal);
-      expectedSecondBasal.suppressed = expectedSecondBasal.suppressed
-        .set('scheduleName', 'billy').done();
-      expectedSecondBasal = expectedSecondBasal.done();
-      var expectedThirdBasal = _.cloneDeep(regBasal2);
-      expectedThirdBasal = expectedThirdBasal.set('duration', 19200000)
-        .set('previous', _.omit(expectedSecondBasal, 'previous'))
-        .done();
-      expectedThirdBasal.annotations = [{code: 'insulet/basal/fabricated-from-schedule'}];
-      thisSim.basal(regBasal1);
-      thisSim.basal(tempBasal);
-      thisSim.basal(regBasal2);
-      thisSim.finalBasal();
-      expect(thisSim.getEvents()).deep.equals([
-        expectedFirstBasal,
-        expectedSecondBasal,
-        expectedThirdBasal
+        .with_duration(180000)
+        .with_rate(1.3);
+      var suppressed2 = builder.makeScheduledBasal()
+        .with_time('2014-09-25T18:30:00.000Z')
+        .with_deviceTime('2014-09-25T18:30:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_duration(180000)
+        .with_rate(2);
+      var tempBasal = builder.makeTempBasal()
+        .with_time('2014-09-25T18:15:00.000Z')
+        .with_deviceTime('2014-09-25T18:15:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0);
+      var tempBasal2 = builder.makeTempBasal()
+        .with_time('2014-09-25T18:30:00.000Z')
+        .with_deviceTime('2014-09-25T18:30:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0);
+      var tempBasalStart = {
+            type: 'temp-basal',
+            subType: 'start',
+            percent: 0.65,
+            duration: 1800000
+          };
+      var tempBasalStop = {
+            type: 'temp-basal',
+            subType: 'stop',
+            time_left: 0
+          };
+      var basal3 = builder.makeScheduledBasal()
+        .with_time('2014-09-25T19:00:00.000Z')
+        .with_deviceTime('2014-09-25T19:00:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_rate(2);
+
+      var expectedTempBasal = tempBasal.with_payload({duration:1800000})
+                                        .set('percent',0.65)
+                                        .with_duration(900000)
+                                        .with_previous(suppressed1.done())
+                                        .done();
+      var expectedTempBasal2 = tempBasal2.with_payload({duration:1800000})
+                                        .set('percent',0.65)
+                                        .with_duration(900000)
+                                        .with_previous(suppressed2.done())
+                                        .done();
+
+      simulator.basal(suppressed1);
+      simulator.tempBasal(tempBasalStart);
+      simulator.basal(tempBasal);
+      simulator.basal(suppressed2);
+      simulator.basal(tempBasal2);
```

Comment:
To check my own understanding here I threw in some logging while uploading to look at a temp crossing schedule boundary:
```
basal called in simulator 2015-12-01T13:00:01 ["timed_segment"]
tempBasal called in simulator undefined start
basal called in simulator 2015-12-02T03:44:15 ["temp_rate_start"]
basal called in simulator 2015-12-02T04:04:15 ["timed_segment"]
basal called in simulator 2015-12-02T06:44:15 ["temp_rate_end"]
tempBasal called in simulator undefined stop
basal called in simulator 2015-12-02T08:04:15 ["timed_segment"]
```

We're not building anything from a change_type `temp_rate_end` so we do indeed have the equivalents of a `suppressed1`, `tempBasalStart`, `tempBasal`, `tempBasal2`, `tempBasalStop`, and `basal3`, in that order. So remove the `suppressed2` plz?